spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/embedded/EmbeddedLdapAutoConfiguration.java

```diff
@@ -64,7 +65,6 @@
 @EnableConfigurationProperties({ LdapProperties.class, EmbeddedLdapProperties.class })
 @AutoConfigureBefore(LdapAutoConfiguration.class)
 @ConditionalOnClass(InMemoryDirectoryServer.class)
-@ConditionalOnProperty(prefix = "spring.ldap.embedded", name = "base-dn")
```

Comment:
I think that's pretty bad. We should let the condition infra does its job  rather than failing as you've done instead.

Reply:
Yes, it is :cry: Using the following format `spring.ldap.embedded.base-dn[0]:dc=spring,dc=org` doesn't match with `@ConditionalOnProperty(prefix = "spring.ldap.embedded", name = "base-dn")`